src/gui/qgsbrowserdockwidget.h

```diff
@@ -102,6 +102,8 @@ class GUI_EXPORT QgsBrowserDockWidget : public QgsDockWidget, private Ui::QgsBro
     void openFile( const QString & );
     //! Emitted when drop uri list needs to be handled
     void handleDropUriList( const QgsMimeDataUtils::UriList & );
+    //! Connections changed in the browser
+    void connectionsChanged( );
```

Comment:
@wonder-sk I've found some time for an initial base class implementation and a test WMS implementation of the synchronization connections. It looks likes it's working as expected and synchronization works fine in both directions.

Would you please confirm that this is a good solution (for now WMS only!) before I spend more time refactoring the other source select widgets and data items?

I'm not convinced about the naming of the new class, I would have called it `QgsSourceSelectDialog` but that's already taken by the class that @manisandro  wrote for the ARC* providers. 

@manisandro if that class was not intended to be the real base class for all source select dialogs (it appears to be too specialized for online services), would you mind if I rename it to something more in line with its functionality as a base class for ARC* online services only? (suggestions welcome)